packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -113,6 +117,18 @@ class HotRunner extends ResidentRunner {
     for (FlutterDevice device in flutterDevices)
       device.initLogReader();
 
+    if (previewDart2 && (generator == null)) {
+      generator = new ResidentCompiler(
+        artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath));
+      final String outputFile = await generator.compile(mainPath);
+      // If compilation was successful, no reason VM should reject it.
```

Comment:
Generally speaking, bad code should be flagged and diagnosed by the frontend I believe. But whatever kernel binary frontend produces I assume VM should be able to load. I think it's good initial assumption at least :-)